File: food-delivery-app/lib/features/authentication/models/account/setting.dart
import 'package:food_delivery_app/features/authentication/models/account/user.dart';
import 'package:food_delivery_app/utils/helpers/helper_functions.dart';

class UserSetting {
  final User? user;
  final bool? notification;
  final bool? darkMode;
  final bool? sound;
  final bool? automaticallyUpdated;
  final String? language;

  UserSetting({
    required this.user,
    required this.notification,
    required this.darkMode,
    required this.sound,
    required this.automaticallyUpdated,
    required this.language,
  });

  UserSetting.fromJson(Map<String, dynamic> json)
      : user = User.fromJson(json['user']),
        notification = json['notification'],
        darkMode = json['dark_mode'],
        sound = json['sound'],
        automaticallyUpdated = json['automatically_updated'],
        language = json['language'];

  Map<String, dynamic> toJson() {
    return {
      'user': user?.toJson(),
      'notification': notification,
      'dark_mode': darkMode,
      'sound': sound,
      'automatically_updated': automaticallyUpdated,
      'language': language,
    };
  }

  @override
  String toString() {
    return THelperFunction.formatToString('UserSetting', {
      'user': user,
      'notification': notification,
      'darkMode': darkMode,
      'sound': sound,
      'automaticallyUpdated': automaticallyUpdated,
      'language': language,
    });
  }
}

class UserSecuritySetting {
  final UserSetting? setting;
  final bool? faceId;
  final bool? touchId;
  final bool? pinSecurity;

  UserSecuritySetting({
    required this.setting,
    required this.faceId,
    required this.touchId,
    required this.pinSecurity,
  });

  UserSecuritySetting.fromJson(Map<String, dynamic> json)
      : setting = UserSetting.fromJson(json['setting']),
        faceId = json['face_id'],
        touchId = json['touch_id'],
        pinSecurity = json['pin_security'];

  Map<String, dynamic> toJson() {
    return {
      'setting': setting?.toJson(),
      'face_id': faceId,
      'touch_id': touchId,
      'pin_security': pinSecurity,
    };
  }

  @override
  String toString() {
    return THelperFunction.formatToString('UserSecuritySetting', {
      'setting': setting,
      'faceId': faceId,
      'touchId': touchId,
      'pinSecurity': pinSecurity,
    });
  }
}

File: food_delivery_backend/account/models/setting.py
import uuid
from django.db import models

class Setting(models.Model):
    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False, db_index=True)
    user = models.OneToOneField("account.User", on_delete=models.CASCADE, related_name="setting", db_index=True)
    notification = models.BooleanField(default=True)
    dark_mode = models.BooleanField(default=False)
    sound = models.BooleanField(default=False)
    automatically_updated = models.BooleanField(default=False)
    LANGUAGE_CHOICES = (
        ("ENGLISH", "English"),
        ("VIETNAMESE", "Vietnamese"),
    )
    language = models.CharField(max_length=30, choices=LANGUAGE_CHOICES, default="ENGLISH")

class SecuritySetting(models.Model):
    setting = models.OneToOneField("account.Setting", on_delete=models.CASCADE, related_name="security_setting", db_index=True)
    face_id = models.BooleanField(default=False)
    touch_id = models.BooleanField(default=False)
    pin_security = models.BooleanField(default=False)
